openvpn/session/manager.go

```diff
@@ -1,36 +1,44 @@
 package session
 
 import (
+	"github.com/mysterium/node/identity"
 	"github.com/mysterium/node/openvpn"
 	"github.com/mysterium/node/session"
+	"sync"
 )
 
-func NewManager(clientConfig *openvpn.ClientConfig) *manager {
+// NewManager returns session manager which maintans a map of session id -> session
+func NewManager(clientConfig *openvpn.ClientConfig, idGenerator session.Generator) *manager {
 	return &manager{
-		idGenerator:  &session.Generator{},
+		idGenerator:  idGenerator,
 		clientConfig: clientConfig,
-		sessions:     make([]session.SessionID, 0),
+		sessionMap:   make(map[session.SessionID]session.Session),
+		creationLock: &sync.Mutex{},
 	}
 }
 
 type manager struct {
-	idGenerator  session.GeneratorInterface
+	idGenerator  session.Generator
 	clientConfig *openvpn.ClientConfig
-	sessions     []session.SessionID
+	sessionMap   map[session.SessionID]session.Session
+	creationLock *sync.Mutex
```

Comment:
This mutex doesn't have to be pointer - it can be just `sync.Mutex`.

Reply:
Fixed